server/src/main/java/io/druid/server/AsyncQueryForwardingServlet.java

```diff
@@ -78,6 +80,7 @@
   private static final String HOST_ATTRIBUTE = "io.druid.proxy.to.host";
   private static final String SCHEME_ATTRIBUTE = "io.druid.proxy.to.host.scheme";
   private static final String QUERY_ATTRIBUTE = "io.druid.proxy.query";
+  private static final String AVATICA_QUERY_ATTRIBUTE = "io.druid.proxy.avaticaQuery";
```

Comment:
if this an avatica specific routing policy how about doing it in a different Guice module, my question is if somehow we want to add similar functionality for a different tool do I have to add the same thing here as well and add a branch? IMO this can be done by extending this class and bind it via Guice thought.

Reply:
Hm, I feel like making the query handling for the router into an extensible module system is beyond the scope of this PR.

For now, I'd think it's fine if you wanted to add another branch here. What use case did you have in mind?